### DISCUSSION TITLE
Feature Request: Group or similar "Tenant" for stats?

### DISCUSSION THREAD

**Discussion by kurucu** [spatie/laravel-stats#5] [Ideas]
Hi Team,

I've already added the task on my project to incorporate this and remove the stats library I dreamt up!

Would you consider adding some sort of grouping column/attribute? For example, to set the stats per tenant, user, project or similar? 

E.

**Comment by fefo-p** [comment]
Would be a really nice feature to have

**Comment by adrianspacely** [comment]
I'd prefer a config file to specify my own StatsEvent model. Then I can either add my own scope or specify a tenant connection.

**Reply by adrianspacely** [reply]
This worked for me. I added a `tenant_id` column to the `stats_events` table and did the following in my `AppServiceProvider` to scope everything to my selected tenant:

```
StatsEvent::creating(function ($model) {
    $model->tenant_id = optional(tenant())->id;
});
StatsEvent::addGlobalScope('tenant', function ($builder) {
    if ($tenant = tenant()) {
        $builder->where('tenant_id', $tenant->id);
    } else {
        $builder->whereNull('tenant_id');
    }
});
```

**Comment by freekmurze** [comment]
We’d accept a PR that adds a ’model’ key to the config file. The specified model should ve used throughout the package.

**Comment by christoph-kluge** [comment]
I did added a PR which adds custom "has-many"-relationship support https://github.com/spatie/laravel-stats/pull/17

```
StatsWriter::for($tenant->stats())

StatsWriter::for($user->stats())->increase(1, ['type' => 'orders'])
StatsQuery::for($user->stats(), ['type' => 'orders'])
    ->start(now()->subMonths(2))
    ->end(now()->subSecond())
    ->groupByWeek()
    ->get(); 
```

**Comment by rainbowdog1** [comment]
Would be fire!
